### PR TITLE
Containerized development environment

### PR DESCRIPTION
**Pull request by mattwang44**
---
name: dev-env
about: Propose a pull request
title: Containerized development environment
labels: dev-env
assignees: TBD

---

# Description

There are too many requisitions (as written in README.md) for newly-joined developers before they start to write/run their code in the local machine. Automating these procedures might help.

- `dev.Dockerfile`: Simplified Dockerfile for containerizing web app (modified from `/Dockerfile`).
- `docker-compose-dev.yml`: Compose file for launching Postgres and web app.
- `docker-entrypoint.sh`: The script activated when the web app container is launched with docker-compose for the first time, including db migration & create super user in Django.
- `enter_dev_ens.sh`: Script for developers (users) to activate the development environment.

Reference: https://kkc.github.io/2018/04/28/docker-note/

# Steps to Test This Pull Request

Follow the steps in `contrib/dev_env.md`:

1. Update `DATABASE_URL` in `src/pycontw2016/settings/local.env`
    ```
    DATABASE_URL=postgres://postgres:secretpostgres@db:5432/pycontw2016
    ```
2. Run the activation script
    ```
    ./enter_dev_env.sh
    ```

# Expected behavior

`localhost:8000` displays the web page successfully (same with following steps in README.md)

# More Information

 - The server launched inside the container. It may lead to increased latency when displaying the web page.
- Might need additional efforts to check if every operation we used to have still stays functional if we adopt this method.

### CODE REVIEW

File: contrib/docker-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+echo 'Run migration'
+python3 /app/src/manage.py migrate
+echo 'Create super user'
+python3 /app/src/manage.py createsuperuser --noinput || echo "Super user already created"
+exec "$@"
```
Comment (tai271828) [inline contrib/docker-entrypoint.sh:6]: nitpicking:   no newline at end of file

Comment (mattwang44) [reply contrib/docker-entrypoint.sh]: updated. thanks!

**Comment by tai271828** [comment]
I tried this pull request by following the instruction. It looks so good to me! I am looking forward to this feature for a such while. Thanks @mattwang44 . I think many developers will be benefited from your contribution as well. Bravo!

```
  Applying users.0010_cocrecord... OK
  Applying users.0011_cocrecord_agreed_at... OK
Create super user
Superuser created successfully.
[04/Jun/2020 21:49:49] I Watching for file changes with StatReloader
[04/Jun/2020 21:49:49] I Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
June 04, 2020 - 21:49:50
Django version 3.0.3, using settings 'pycontw2016.settings.local'
Starting development server at http://0.0.0.0:8000/
Quit the server with CONTROL-C.
```

**Comment by tai271828** [comment]
I suppose this pull request will conflict with https://github.com/pycontw/pycon.tw/pull/791 . Let's merge this one first to make code-landing easier. I will fix the conflict from https://github.com/pycontw/pycon.tw/pull/791 later.

**Pull request merged**